dev/com.ibm.ws.ejbcontainer.core/test/com/ibm/ejs/container/ScheduleExpressionParserTest.java

```diff
@@ -2088,53 +1994,226 @@ public void testStartAndEnd()
+            throw new Error(t2009_11_01__02b_00_00m + " " + millisToString(t2009_11_01__03_00_00m));
+        }
+        if (!millisToString(t2009_11_01__03_00_00m, timezone).equals("2009-11-01 03:00:00")) {
+            throw new Error(millisToString(t2009_11_01__03_00_00m, timezone));
+        }
     }
 
     @Test
-    public void testExamples()
-    {
+    public void testDaylightSavingsTimeFirstTimeout() throws Exception {
+        // This test relies on US-specific DST transitions in 2009.
+
+        // Note: getFirstTimeout uses current system time, so instead test the
+        // method called by getFirstTimeout, getTimeout(time, false);
+
+        // First, cover springing ahead to DST; specific hours, every hour, and every minute
+        final long t2009_03_08__00_59_00 = stringToMillis("2009-03-08 00:59:00 America/Chicago");
+        verifyFirstTimeout(new ScheduleExpression().hour("0-4").timezone("America/Chicago"),
+                           t2009_03_08__00_59_00,
+                           t2009_03_08__00_59_00 + (60 * 1000),
+                           "2009-03-08 01:00:00 America/Chicago");
+
+        final long t2009_03_08__01_59_00 = t2009_03_08__00_59_00 + (1 * 60 * 60 * 1000);
+        verifyFirstTimeout(new ScheduleExpression().hour("0-4").timezone("America/Chicago"),
+                           t2009_03_08__01_59_00,
+                           t2009_03_08__01_59_00 + (60 * 1000),
+                           "2009-03-08 03:00:00 America/Chicago");
+        verifyFirstTimeout(new ScheduleExpression().hour("*").timezone("America/Chicago"),
+                           t2009_03_08__01_59_00,
+                           t2009_03_08__01_59_00 + (60 * 1000),
+                           "2009-03-08 03:00:00 America/Chicago");
+        verifyFirstTimeout(new ScheduleExpression().hour("*").minute("*").timezone("America/Chicago"),
+                           t2009_03_08__01_59_00,
+                           t2009_03_08__01_59_00,
+                           "2009-03-08 01:59:00 America/Chicago");
+
+        final long t2009_03_08__01_59_01 = t2009_03_08__01_59_00 + 1000;
+        verifyFirstTimeout(new ScheduleExpression().hour("*").timezone("America/Chicago"),
+                           t2009_03_08__01_59_01,
+                           t2009_03_08__01_59_01 + (59 * 1000),
+                           "2009-03-08 03:00:00 America/Chicago");
+        verifyFirstTimeout(new ScheduleExpression().hour("*").minute("*").timezone("America/Chicago"),
+                           t2009_03_08__01_59_01,
+                           t2009_03_08__01_59_01 + (59 * 1000),
+                           "2009-03-08 03:00:00 America/Chicago");
+
+        // Second, cover falling back from DST; specific hours, every hour, and every minute
+        final long t2009_11_01__00_00_00 = stringToMillis("2009-11-01 00:00:00 America/Chicago");
+        verifyFirstTimeout(new ScheduleExpression().hour("*").timezone("America/Chicago"),
```

Comment:
should some of these be `new ScheduleExpression().hour("0-4")` instead of wildcard like above for spring ahead?

Reply:
Added multiple new combinations: hour("1"), hour("0-2").minute("*).second("*"), hour("1,3")